### ISSUE TITLE
Customized inputAccessoryView not taking consideration of safeArea on iPhone X

### ISSUE THREAD

**Issue by hyouuu** [MessageKit/MessageKit#357]
## General Information
In my app I replace the text input bar with a "Login" button when a user is not logged in by overriding inputAccessoryView - something like:
```
open var inputAccessoryView: UIView? { return [a UIButton, for example] }
```
It appears fine on normal iPhone, but for iPhone X, it sticks to the bottom part and looks weird.

* MessageKit Version:
10.12 development head

* iOS Version(s):
11.1.2

* Swift Version:
4 with Xcode 9

* Devices/Simulators:
iPhone X

* Reproducible in ChatExample? (Yes/No):
No

**Comment by SD10** [comment]
@hyouuu You might want to take a look at this PR and see if it is of any use https://github.com/MessageKit/MessageKit/pull/350/files

**Comment by hyouuu** [comment]
Mmm actually have you tested that PR out? Before updating `development` branch the normal text input bar was working correctly, and now it is also broken T.T

**Comment by hyouuu** [comment]
@SD10 in case

**Comment by SD10** [comment]
Yes, of course, I've tested it or I would never allow it to be merged 🤔 I don't recommend you use the development branch though.

Can you provide a screenshot of what you're experiencing?

**Comment by hyouuu** [comment]
Sure thing - before switching to development it looked like this with message input bar, which is fine-ish (the double bar was my tab bar leaking out):
![before](https://user-images.githubusercontent.com/5061867/33056396-507654c0-ce3a-11e7-9ba8-b6879ca526ec.png)

After switching to development with that patch, now it apparently doesn't consider safe area at all:
![after](https://user-images.githubusercontent.com/5061867/33056395-50603424-ce3a-11e7-9e84-5af4660e8ce7.png)

When replacing inputAccessoryView with another view either before or after:
![replace](https://user-images.githubusercontent.com/5061867/33056397-508c8b5a-ce3a-11e7-956e-2699bceb2c0c.png)

I'm trying to start contributing to MessageKit a bit, so used development :D

**Comment by hyouuu** [comment]
@SD10

**Comment by hyouuu** [comment]
That PR moved safeArea related constraint to didMoveToWindow(), and it looks problematic since didMoveToWindow sounds like is only called when focusing?

**Comment by nathantannar4** [comment]
We needed to move it to didMoveToWindow otherwise it caused a crash. However if you replace and override the inputaccessory view you will need to anchor it on your own.

**Comment by hyouuu** [comment]
Thanks for the explanation. But as the screenshot shows, even the normal message input bar doesn't work (and if bringing up keyboard the bar will be half blocked)

**Comment by nathantannar4** [comment]
I think You need to hide the tab bar. The inputAccessoryView is always laid overtop

**Comment by hyouuu** [comment]
Have you tried the latest head in `development` on iPhone X? As the 2nd screenshot shows, it is apparently not using safeArea's bottom. Here is a more apparent one with keyboard up:

![img_0127](https://user-images.githubusercontent.com/5061867/33102781-47433eba-ced3-11e7-9d40-17ba7a4c50d1.png)

**Comment by nathantannar4** [comment]
Is this reproducible in the Example? 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by hyouuu** [comment]
Yes.

<img width="432" alt="screen shot 2017-11-22 at 12 33 22 pm" src="https://user-images.githubusercontent.com/5061867/33148634-6b70978e-cf81-11e7-89eb-95c25199f376.png">
<img width="438" alt="screen shot 2017-11-22 at 12 33 31 pm" src="https://user-images.githubusercontent.com/5061867/33148635-6b889f14-cf81-11e7-9611-a4a8cc39c0e3.png">

**Comment by nathantannar4** [comment]
Note: This issue is not present on the `master` branch

**Comment by SD10** [comment]
This should be resolved once #358 is merged in. I'm going to close this issue because it is related to a development branch. But thank you for catching this early @hyouuu 😅
